### PR TITLE
Change `CoreUpgrader::download_package` signature

### PR DESCRIPTION
**Pull request by wojsmol**
Fixes https://github.com/wp-cli/core-command/issues/167

### CODE REVIEW

File: src/WP_CLI/Core/CoreUpgrader.php [hunk 1]
```diff
@@ -25,9 +25,10 @@ class CoreUpgrader extends DefaultCoreUpgrader {
 	 * @param string $package          The URI of the package. If this is the full path to an
 	 *                                 existing local file, it will be returned untouched.
 	 * @param bool   $check_signatures Whether to validate file signatures. Default true.
+	 * @param array  $hook_extra       Extra arguments to pass to the filter hooks. Default empty array.
 	 * @return string|WP_Error The full path to the downloaded package file, or a WP_Error object.
 	 */
-	public function download_package( $package, $check_signatures = true ) {
+	public function download_package( $package, $check_signatures = true, $hook_extra = array() ) {
```
Comment (schlessera) [inline src/WP_CLI/Core/CoreUpgrader.php:31]: Short array syntax is used for WP-CLI.
```suggestion
	public function download_package( $package, $check_signatures = true, $hook_extra = [] ) {
```

**Comment by schlessera** [comment]
How is this called and what is the behavior on older Core versions? I'm wondering whether we might need to support two different versions of the signature to maintain BC...

**Comment by schlessera** [comment]
The `$hook_extra` argument actually needs to be forwarded to the `'upgrader_pre_download'` filter within the method as well.

**Review by schlessera — APPROVED** [review]

**Comment by TimothyBJacobs** [comment]
This causes warnings on PHP 7.4 and fatals on PHP 8.0 for any installs on < WP 5.5. Originally reported here: https://github.com/wp-cli/wp-cli/issues/5452

> I'm wondering whether we might need to support two different versions of the signature to maintain BC...

I think this will be necessary.

**Comment by webdock-io** [comment]
Please see https://github.com/wp-cli/wp-cli/issues/5494 - This seems related and still an issue, somehow

**Pull request merged**